src/arrayMethodJoin.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    let spr = ',';
+
+    if (typeof separator !== 'undefined') {
+      spr = String(separator);
+    }
+
+    for (let ind = 0; ind < this.length; ind++) {
+      let element = '';
+
+      if (!(this[ind] === null || typeof this[ind] === 'undefined')) {
+        element = String(this[ind]);
+      }
+
+      if (ind === this.length - 1) {
+        result = result.concat(element);
+        break;
+      }
+
+      result = result.concat(element, spr);
```

Comment:
```
      if (this[i] !== undefined && this[i] !== null) {
        result += `${this[i]}`;
      }

      if (i < this.length - 1) {
        result += separator;
      }
```
you need only this here
it will be simplier